### PR TITLE
[ext] Update all submodules

### PR DESCRIPTION
**Pull request by salkinium**
Let's update the submodules:

- [x] TinyUSB changes tested in hardware.
  - [x] DFU callback changed, added explicity delay argument to `scons|make program-dfu delay={seconds}`
  - [x] Midi write24 signature changed, Midi class depends on Audio class
  - [x] Improved partial customizaton support via weakly linked descriptors
- [x] FatFs changes tested in hardware.
- [x] Added Petit FatFs module
- [x] FreeRTOS changes tested in hardware.
- [x] FreeRTOS-TCP changes tested in hardware.
- [x] Better configuration management with `#ifndef`-guarded modm defaults.

cc @rleh @chris-durand

### CODE REVIEW

**Comment by rleh** [comment]
Like [dependabot](https://github.com/marketplace/dependabot-preview), but for submodules?

**Comment by salkinium** [comment]
Ok, so I'm not fixing this any time soon, I just wanted to know if updating modm-devices causes any issues. But apparently #390 has some coding bugs unrelated to modm-devices data.

**Comment by salkinium** [comment]
I changed the `update.py` scripts for TinyUSB, CMSIS and FreeRTOS to only copy the latest tagged release. This is necessary to not pick some development version that's buggy.

**Comment by salkinium** [comment]
Note to self: Refactor submodules to use GH Actions for the update code with proper git commit message like: https://github.com/modm-ext/etl-partial

**Comment by salkinium** [comment]
All partial repositories have been refactored to use GHA and much better commit messages.

**Comment by salkinium** [comment]
@rleh Would you test the FreeRTOS+TCP example? I don't have an Ethernet-USB adapter around…

**Comment by rleh** [comment]
> @rleh Would you test the FreeRTOS+TCP example? I don't have an Ethernet-USB adapter around…

Tested with the updated submodules from you branch:

```
[rleh@rleh-pc ethernet]$ ping 192.168.1.1
PING 192.168.1.1 (192.168.1.1) 56(84) bytes of data.
64 Bytes von 192.168.1.1: icmp_seq=1 ttl=64 Zeit=0.568 ms
64 Bytes von 192.168.1.1: icmp_seq=2 ttl=64 Zeit=0.136 ms
[...]

[rleh@rleh-pc ethernet]$ curl -v 192.168.1.1
*   Trying 192.168.1.1:80...
* Connected to 192.168.1.1 (192.168.1.1) port 80 (#0)
> GET / HTTP/1.1
> Host: 192.168.1.1
> User-Agent: curl/7.69.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK 
< Content-Type: text/html
< Connection: keep-alive
* no chunk, no close, no size. Assume close to signal end
< 
* Closing connection 0
<html><body><h1>Hello from your STM32!</h1></body></html>
```
 Looks good to me 👍🏽

File: tools/build_script_generator/make/resources/Makefile.in [hunk 1]
```diff
@@ -87,9 +87,10 @@ program: build
 program-bmp: build
 	@python3 modm/modm_tools/bmp.py -p $(port) $(ELF_FILE)
 
+delay?=0
```
Comment (rleh) [inline tools/build_script_generator/make/resources/Makefile.in]:
```suggestion
delay?=5
```

**Comment by salkinium** [comment]
I changed the FatFs config to be in line with the LVGL config. I added a Petit FatFs module, but without example (yet).

I will next check what we can do about the FreeRTOS config which I think should also get the same treatment for consistency. (Shouldn't be too difficult).

**Comment by salkinium** [comment]
This is ready for review and merging.

File: ext/aws/freertos_tcp.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# FreeRTOS+TCP
+
+Amazon FreeRTOS+TCP is a scalable, open source and thread safe TCP/IP stack for
+FreeRTOS. FreeRTOS+TCP's features and RAM footprint are fully scalable, making
+it equally applicable to smaller lower throughput microcontrollers as larger
+higher throughput microprocessors.
+
+This module provides the latest FreeRTOS+TCP LTS release intergrated with modm:
```
Comment (rleh) [inline ext/aws/freertos_tcp.md]:
```suggestion
This module provides the latest FreeRTOS+TCP LTS release integrated with modm:
```

File: ext/aws/freertos.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Configuration
+
+This module generates a `FreeRTOSConfig.h` config file for modm integration with
+these settings:
+
+- `configASSERT(x)` implemented with `modm_assert(x, "freertos")`.
+- `configCPU_CLOCK_HZ` implemented with CMSIS `SystemCoreClock`.
+- `configTICK_RATE_HZ` set to `modm:freertos:frequency` or 1kHz on Cortex-M0.
+- `configSUPPORT_DYNAMIC_ALLOCATION = 1` as implemented by `modm:platform:heap`.
```
Comment (rleh) [inline ext/aws/freertos.md:28]: Is there a reason why this cannot be a user setting?
(and the dependency on `modm:platform:heap` could be removed)

**Review by rleh — COMMENTED** [review]
Nice 👍🏽

**Review by chris-durand — APPROVED** [review]

File: ext/aws/freertos.md (hunk 3 above)
Comment (salkinium) [reply ext/aws/freertos.md:28]: It was like that before 😇. I didn't change it ;-P
You can force it by `#undef`ing the definition in the local config. However, I'm not sure how well FreeRTOS is usable without heap. At least the `modm:processing:rtos` code requires dynamic heap calls to `xSemaphoreCreateMutex` (vs. `xSemaphoreCreateMutexStatic`), `xQueueCreate`, `xTaskCreate` etc.

**Review by rleh — APPROVED** [review]

**Pull request merged**